Config handler increment tool version

eliasyishak opened the pull request:
This new feature will allow for a newer version of the message to increment the version number in the config file

eliasyishak committed:
added new method for incrementing version + simplified getter in analytics

eliasyishak committed:
added getter to get parsed tools in config

eliasyishak committed:
Merge pull request #12 from eliasyishak/main

Merging refactor of messaging logic to keep messages inside package

eliasyishak committed:
added test for incrementing tools version

Jasguerrero commented on lib/src/config_handler.dart:
```diff
@@ -102,6 +102,43 @@ class ConfigHandler {
+
+    // TODO: need to determine what to do when there are two lines for the same tool
+    //  as outlined in the Dev Notes document; currently only assuming one line per tool
+    if (matches.length == 1) {
```
Could we raise an exception as not implemented when `matches.length > 1` and a test for that?

eliasyishak replied:
Great idea, I actually should just implement the method that will get triggered when the matches aren't equal to 1 so I can get rid of this TODO

eliasyishak replied:
Just added the functionality to reset the configuration file if that conditional fails. Also added tests for it

eliasyishak committed:
logic to increment tool if tool already exists and is stale version

eliasyishak committed:
removing conditional

This conditional isn't necessary because that conditional will always fail due to the conditional in analytics.dart that triggered it

eliasyishak committed:
updating with annotation for toolsMessage member

eliasyishak committed:
adding functionality to reset the config file if regex matches error out

eliasyishak committed:
tests added to confirm that reset works in both cases

case 1 -- when there is not exactly one match for the reporting flag line

case 2 -- when there is not exactly one match for a tool when incrementing the version

eliasyishak committed:
clean up

eliasyishak committed:
Update constants.dart

Jasguerrero reviewed (approved):
LGTM
